standard/standard-library.md

```diff
@@ -396,6 +407,164 @@ namespace System
         public OperationCanceledException(string? message, Exception? innerException);
     }
 
+    /// <summary>
+    ///    A read-only value type which represents an abstract
+    ///    index to be used with collections.
+    ///    - The Index can be relative to the start or end of a
+    ///      collection.
+    ///    - An Index can be converted to a zero-based concrete
+    ///      from-start index to be used with a collection
+    ///      of some specified length.
+    ///    - Equality between Index values is provided, however
+    ///      unlike concrete indicies they are not ordered.
+    ///    - Array and String element access support indexing
+    ///      with Index values.
+    /// </summary>
+    public readonly struct Index : IEquatable<Index>
+    {
+        /// <summary>
+        ///    Construct an Index from an integer value and a
+        ///    boolean indicating whether the value is relative
+        ///    to the end (true) or start (false).
+        /// </summary>
+        /// <param name="value">
+        ///    The value, must be ≥ 0.
+        /// </param>
+        /// <param name="fromEnd">
+        ///    Optional boolean indicating whether the Index is
+        ///    relative to the end (true) or start (false).
+        ///    The default value is false.
+        /// </param>
+        /// <exception cref="ArgumentOutOfRangeException">
+        ///    Thrown if value < 0.
+        /// </exception>
+        /// <remarks>
+        ///    If the Index is relative to the start then:
+        ///       - the value 0 refers to the first element.
+        ///    If the Index is relative to the end then:
+        ///       - the value 1 refers to the last element; and
+        ///       - the value 0 refers to beyond last element.
+        /// </remarks>
+        public Index(int value, bool fromEnd = false);
+
+        /// <summary>
+        ///    Implicit conversion from integer to a
+        ///    from-start Index.
+        /// </summary>
+        /// <remarks>
+        ///    The predefined operator:
+        ///       <c>Index operator ^(int value);</c>
+        ///    is provided to convert from integer to a
+        ///    from-end Index.
+        /// </remarks>
+        public static implicit operator Index(int value);
+
+        /// <summary>
+        ///    Return the value.
+        /// </summary>
+        public int Value { get; }
+
+        /// <summary>
+        ///    Return whether the Index is relative to
+        ///    the end (true) or start (false).
+        /// </summary>
+        public bool IsFromEnd { get; }
```

Comment:
I did *not* include this property in my proposal. Should it be included?

Reply:
So folk know my reasoning, for both `Index` & `Range` I included:

- the constructor – these are `readonly` types with values set on construction;
- properties that return the type's values – so they can be used;
- `IEquatable`’s `Equals` – so they can be compared; and
- the method to compute a concrete offset (& length for `Range`) – so the math is done for you ;-)

and nothing else that a certain implementation may offer as their value-add.